src/api/lib/xpath_engine.rb

```diff
@@ -45,9 +45,9 @@ def initialize
         'attribute_issue/owner/@email' => {cpart: 'users.email', joins:           ['LEFT JOIN users ON users.id = attribissues.owner_id']},
         'attribute_issue/owner/@login' => {cpart: 'users.login', joins:           ['LEFT JOIN users ON users.id = attribissues.owner_id']},
         'person/@userid'               => {cpart: 'users.login', joins:           ['LEFT JOIN users ON users.id = user_relation.user_id']},
-        'person/@role'                 => {cpart: 'ppr.title', joins:           ['LEFT JOIN roles AS ppr ON user_relation.role_id = ppr.id']},
```

Comment:
does nobody else find this indentation strange? 🤔

Reply:
It would be nicer to align the values. But it doesn't seem to the enabled cop(s) are responsible for checking for this offense. Otherwise this would fail.

Reply:
So, hash alignment is checked by [Layout/AlignHash](http://rubocop.readthedocs.io/en/latest/cops_layout/#layoutalignhash). That cop is already enabled.

```ruby
# good
{
  foo: bar,
  ba: baz
}
{
  :foo => bar,
  :ba => baz
}
```

And if I understand correctly [Layout/ExtraSpacing](http://rubocop.readthedocs.io/en/latest/cops_layout/#layoutextraspacing) would allow different spacing.

It doesn't seem that rubocop currently has a cop that would enforce spacing for cases like the one you found. Don't get me wrong, I agree that it would be nicer to align it, and I assume noone is against it. But as for right now it seems independent from any rubocop rules ([Layout/AlignHash](http://rubocop.readthedocs.io/en/latest/cops_layout/#layoutalignhash) would affect it, but in a different way).

Reply:
Of course that `Layout/AlignHash` won't complain about this. Because this is about the indentantion of things in different hashes. You always have to review what Rubocop autocorrect does, sometimes the alternative its provides is not the best one.

A much clearer example (with a different cop) would be:
``` Ruby
return "#{word}"
```
which Rubocop would autocorrect to:
``` Ruby
return word.to_s
```

But if you are sure that `word` is a string (for example because that is a field from the database), you should actually write
``` Ruby
return word
```

Rubocop autocorrection was fine, as it solves the offense, but it is not what we want. And of course there is no cop for this.


So basically there are two ways to correct this:

## 1
``` Ruby
'person/@userid'               => {cpart: 'users.login', joins:           ['LEFT JOIN users ON users.id = user_relation.user_id']},
'person/@role'                 => {cpart: 'ppr.title', joins:             ['LEFT JOIN roles AS ppr ON user_relation.role_id = ppr.id']},
```

## 2
``` Ruby
'person/@userid'               => {cpart: 'users.login', joins:           ['LEFT JOIN users ON users.id = user_relation.user_id']},
'person/@role'                 => {cpart: 'ppr.title', joins:             ['LEFT JOIN roles AS ppr ON user_relation.role_id = ppr.id']},
```
 
Both are right for the Rubocop cop, and it doesn't know what you want, it takes 1. But it is clear we want 2. So I would like to correct this manually to have 2. Because if we are going to correct things better doing it in the best way.


Rubocop autocorrect helps you, but you always have to review what he does: 😉

Reply:
I already told you that I agree that aligning the hash values would be better.


I would prefer 3:

**3**
```ruby
'person/@userid'               => {cpart: 'users.login', joins:           ['LEFT JOIN users ON users.id = user_relation.user_id']},
'person/@role'                 => {cpart: 'ppr.title',   joins:           ['LEFT JOIN roles AS ppr ON user_relation.role_id = ppr.id']},
```

> Ana06 14 minutes ago
> Rubocop autocorrect helps you, but you always have to review what he does.

I already pointed out that this part is not defined by any rubocop rule / the rubocop styleguide;-)

> If Rubocop allow several option for the same rule is our responsability to take the best one. Please do
not introduce more "not beautiful" code just because Rubocop doesn't complain. :wink:

The point of this PR is to enable more rubocop rule and check if everyone agrees on them. Your concern seems to be about additional style rules (in addition to rubocop). And the whole file needs to be "fixed" to be beautiful.
So I would do this in a separate PR.